src/lib/ldap/bind.c

```diff
@@ -220,7 +220,7 @@ int fr_ldap_bind_async(fr_ldap_connection_t *c,
 
 	el = c->conn->el;
 
-	if (ldap_get_option(c->handle, LDAP_OPT_DESC, &fd) == LDAP_SUCCESS) {
+	if ((ldap_get_option(c->handle, LDAP_OPT_DESC, &fd) == LDAP_SUCCESS) && (fd >= 0)) {
```

Comment:
Could you add a comment here to.  That's very surprising behaviour, but given the quality of the libldap code it doesn't surprise me.